### PR TITLE
Feature/PI-204-bulk_trigger

### PR DESCRIPTION
**Pull request by jaklinger**

### CODE REVIEW

File: src/etl/sds/trigger/bulk/operations.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        super().__init__(
+            self, f"s3://{bucket}/{key} should not exist when using bulk trigger"
+        )
+
+
+def validate_no_changelog_number(s3_client, source_bucket):
+    try:
+        s3_client.head_object(Bucket=source_bucket, Key=CHANGELOG_NUMBER)
+    except ClientError as error:
+        if error.response["Error"]["Code"] != "404":
```
Comment (jaklinger) [inline src/etl/sds/trigger/bulk/operations.py:19]: to not_found var

Comment (jaklinger) [inline src/etl/sds/trigger/bulk/operations.py:19]: 404 -> not_found

File: src/layers/etl_utils/trigger/model.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            init=WorkerKey.EXTRACT, changelog_number=BULK_CHANGELOG_NUMBER, name_="bulk"
+        )
+
+    @classmethod
+    def changelog(cls, changelog_number: str) -> Self:
+        return cls(
+            init=WorkerKey.EXTRACT, changelog_number=changelog_number, name_="changelog"
+        )
+
+    # @classmethod
```
Comment (jameslinnell) [inline src/layers/etl_utils/trigger/model.py:35]: Commented code?

File: src/etl/notify/notify.py [hunk 3]
```diff
@@ -1,2 +1,3 @@
 def handler(event, context):
     print(event)  # noqa: T201
```
Comment (jameslinnell) [inline src/etl/notify/notify.py:2]: Why?

File: src/layers/etl_utils/trigger/model.py (hunk 2 above)
Comment (jaklinger) [inline src/layers/etl_utils/trigger/model.py:35]: add comment

File: src/layers/etl_utils/trigger/model.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    # def retry(cls, init: WorkerKey) -> Self:
+    #     return cls(init=init, changelog_number=None, _name="retry")
+
+    def dict(self):
+        changelog_number = self.changelog_number or "null"
```
Comment (jaklinger) [inline src/layers/etl_utils/trigger/model.py:40]:
```suggestion
        changelog_number = self.changelog_number or json.dumps(None)```

File: src/layers/etl_utils/worker/model.py [hunk 5]
```diff
@@ -27,5 +27,5 @@ class WorkerActionResponse:
     unprocessed_records: list[dict]
     processed_records: list[dict]
     s3_input_path: str
-    s3_output_path: str
+    s3_output_path: str | None = field(default=None)
```
Comment (jaklinger) [inline src/layers/etl_utils/worker/model.py:30]: remove

**Pull request merged**